SnowplowTracker/SnowplowTracker/Storage/InMemoryEventStore.cs

```diff
@@ -0,0 +1,128 @@
+﻿/*
+ * InMemoryEventStore.cs
+ * SnowplowTracker.Storage
+ * 
+ * Copyright (c) 2015-2023 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using SnowplowTracker.Payloads;
+
+namespace SnowplowTracker.Storage
+{
+    /// <summary>
+    ///  Buffers events (as TrackerPayloads) in memory before they are sent by the emitter.
+    /// </summary>
+	public class InMemoryEventStore : IStore
+	{
+        private Dictionary<Guid, TrackerPayload> eventQueue = new Dictionary<Guid, TrackerPayload>();
+        private int bufferCapacity;
+
+        /// <summary>
+        /// Initializes a new instance of the event store.
+        /// </summary>
+        /// <param name="bufferCapacity">
+        /// Maximum number of unsent events to hold in memory.
+        /// When crossed, new events won't be added to the store and will be discarded.
+        /// </param>
+		public InMemoryEventStore(int bufferCapacity = 10000)
+		{
+            this.bufferCapacity = bufferCapacity;
+		}
+
+        /// <summary>
+        /// Adds an event payload to the store.
+        /// </summary>
+        /// <returns><c>true</c>, if event was added, <c>false</c> otherwise.</returns>
+        /// <param name="payload">An event payload</param>
+        public bool AddEvent(TrackerPayload payload)
```

Comment:
Admittedly I'm not overly versed in C#, but could this be slightly simplified to remove the need for the bool flag, e.g.:
```
public bool AddEvent(TrackerPayload payload)
{
    lock (eventQueue)
    {
        if (eventQueue.Count < bufferCapacity)
        {
            eventQueue.Add(Guid.NewGuid(), payload);
            Log.Verbose("EventStore: Event added");
            return true;
        }
        else
        {
            Log.Error("EventStore: Reached buffer capacity, event won't be tracked");
            return false;
        }
    }

}
```

Reply:
Yeah, this is a bit cumbersome, but it's actually intentional because I didn't want to do the logging inside the lock. The logging may be a more expensive operation and it doesn't have to be done within a lock, so I think it's more efficient if we do it this way.